# Main configuration for the benchmarking platform
# Please note that all data (different surfaces, tools, etc) should be in the same unit

version: "1.0"
description: "3D coverage path planning benchmarking configuration"
point_density: 50000.0
use_cuda: true

# List of available robot models
robots: 
  - name: "circular"
    type: "cylindrical"
    radius: 0.05 # unit: m
    depth: 0.01 # unit: m

  - name: "line_lidar"
    # TODO: interpolate beam vectors in the code to the desired num
    beam_num: 3 # This might be too many, but we are testing the computing performance
    max_distance: 10 # unit: m
    epsilon: 0.001 # This determines whether a surface point is covered by a beam
    given_beam_vectors: [[0, 0.5, -0.5], [0, -0.5, -0.5], [0, 0, -1]]
    
# List of scenes to test on
scenes: 
  - name: "remeshed_saddle"
    filename: "remeshed_saddle.stl"
    frame_id: "remeshed_saddle"

# List of algorithms
algorithms: 
  - name: "Yang2023Template"